.github/workflows/prevalence.yml

```diff
@@ -1,47 +1,84 @@
 name: Prevalence
 
 on:
-  # schedule:
-  #   # 9:00am Pacific = 16:00 UTC
-  #   - cron:  '0 16 * * *'
+  schedule:
+    # 7:00am Pacific = 14:00 UTC
+    - cron:  '0 14 * * *'
   workflow_dispatch:
 
 jobs:
   prevalence:
-    # TODO(blanchardjeremy): This whole file is a work in progress.
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout git repo (with full history)
+      uses: actions/checkout@v2
+      with:
+          fetch-depth: 0
 
-    - uses: actions/setup-python@v2
+    - name: Set up python
+      uses: actions/setup-python@v2
 
-    - uses: syphar/restore-virtualenv@v1
+    - name: Set up virtualenv
+      uses: syphar/restore-virtualenv@v1
       id: cache-virtualenv
       with:
         requirement_files: requirements-manual.txt
 
-    - uses: syphar/restore-pip-download-cache@v1
+    - name: Set up pip cache
+      uses: syphar/restore-pip-download-cache@v1
       if: steps.cache-virtualenv.outputs.cache-hit != 'true'
 
-      # the package installation will only be executed when the
-      # requirements-files have changed.
-    - run: pip install -r requirements.txt
+    - name: Install python dependencies
+      run: pip install -r requirements-manual.txt
       if: steps.cache-virtualenv.outputs.cache-hit != 'true'
 
-    - name: Run update preavlence
+    - name: Run prevalence update
+      env:
+        CAN_API_KEY: ${{ secrets.COVID_ACT_NOW_API_KEY }}
       run: |
-        CAN_API_KEY=COVID_ACT_NOW_API_KEY python3 update_prevalence.py
+        CAN_API_KEY="$CAN_API_KEY" python3 update_prevalence.py
 
-    # TODO: Delete branch we're going to use in case it exists
 
-    # Checkout a separate branch
-    
-    # Run update_prevalence.py including the CAN API KEY
-    
-    # Run linter (may need to set up all of yarn first?)
+    # We do this after the update_prevalence script in case we are running it from a branch (during testing)
+    - name: Create a branch
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        git stash
+        git checkout main
+        git stash pop
+        git push origin --delete auto-update-prevalence &> /dev/null || true
+        git checkout -b auto-update-prevalence
+        git push --set-upstream origin auto-update-prevalence
 
-    # Commit the updated files with the current date
-    # Create a pull request
-    # Auto merge the pull request
+    # Run linter through directly yarn because it's confifgured properly
+    - name: Setting up yarn
+      uses: borales/actions-yarn@v2.3.0
+      with:
+        cmd: install
+    - name: Running linter fixes
+      uses: borales/actions-yarn@v2.3.0
+      with:
+        cmd: fix
+
+    - name: Get the current date
+      id: date
+      run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+
+    - name: Commit changes
```

Comment:
Is it necessary to use all these modules + steps as opposed to, say

run: yarn update_prevalence -c
env: GITHUB_TOKEN...

Reply:
I thought about that at the beginning. Then I went with this so I didn't have to use my own GitHub personal access token. Then I realized the final merge wouldn't work unless I used a different token so at the very end I went with that anyway. So.... Yeah I could try it? I'd have to give it a closer look to make sure there aren't any other unique aspects to the GitHub version.

Reply:
I'll give it a try on a separate branch to see how it goes. Should be easy now that I know how actions work.

Many of the same modules would need to be loaded up though (python, pip, yarn). We wouldn't need the add-and-commit module, I suppose.